### PR TITLE
patch: Fixing generateUserAgent() to handle null and undefined properly

### PR DESCRIPTION
**Pull request by iAmmarTahir**
This PR aims to handle null and undefined values properly in `generateUserAgent()`

Screenshot 2023-01-30 at 4 24 07 PM

### CODE REVIEW

**Comment by iAmmarTahir** [comment]
@Cole-Greer I have created a new PR with your suggested changes and removed the optional chaining `.?` to support older NodeJS versions. Let me know if there any changes to make.

**Comment by Cole-Greer** [comment]
@iAmmarTahir Thanks that looks great! To get this merged now it will need 3 VOTE +1's from tinkerpop committers. Alternatively if it gets a single VOTE +1 and a week passes, it can be merged with "lazy consensus". You can typically expect some committers to review a PR within a couple of days.

I'm not a tinkerpop committer so I cannot give an official vote here but I will give a VOTE +1 (non-binding) as I think this PR should be accepted.

File: gremlin-javascript/src/main/javascript/gremlin-javascript/lib/utils.js [hunk 1]
```diff
@@ -90,10 +90,24 @@ function generateUserAgent() {
...
-  const osName = os.platform().replace(' ', '_');
-  const osVersion = os.release().replace(' ', '_');
-  const cpuArch = process.arch.replace(' ', '_');
+  let runtimeVersion;
+  let osName;
+  let osVersion;
+  let cpuArch;
+  runtimeVersion = osName = osVersion = cpuArch = 'NotAvailable';
+  if (process != null) {
+    if (process.version != null) {
```
Comment (vkagamlyk) [inline gremlin-javascript/src/main/javascript/gremlin-javascript/lib/utils.js:99]: Can `version` be undefined?
```suggestion
    if (process.version) {
```
same question for `arch`

Comment (iAmmarTahir) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/lib/utils.js:99]: Yes, it makes sense. If the process is not null or undefined, then it must have a version and arch

**Comment by vkagamlyk** [comment]
@iAmmarTahir Thank you for fixing this issue!
VOTE +1

**Pull request merged**